.github/workflows/ci.yml

```diff
@@ -379,12 +381,63 @@ jobs:
           composer config allow-plugins.silverstripe/recipe-plugin true
           composer config allow-plugins.silverstripe/vendor-plugin true
 
-          # Useful to see generated composer.json when diagnosing new bugs
-          cat composer.json
-
           # matrix.composer_args sometimes includes `--prefer-lowest` which is only supported by `composer update`, not `composer install`
           # Modules do not have composer.lock files, so `composer update` is the same speed as `composer install`
-          composer update --no-interaction --no-progress ${{ matrix.composer_args }}
+          # `|| :` prevents an exit code on a failed composer update from halting the workflow
+          composer update --no-interaction --no-progress ${{ matrix.composer_args }} 2> __update_attempt.txt || :
+
+          if ! [[ $(cat __update_attempt.txt) =~ Problem ]]; then
+            # Succesfully ran composer update and installed everything
+            rm __update_attempt.txt
+            # Useful to see generated composer.json when diagnosing new bugs
+            cat composer.json
+          elif ! [[ $(cat __update_attempt.txt) =~ 'Root composer.json requires silverstripe/installer' ]] || \
+            [[ $INPUTS_COMPOSER_REQUIRE_EXTRA =~ silverstripe/installer ]] || \
+            [[ $MATRIX_COMPOSER_REQUIRE_EXTRA =~ silverstripe/installer ]] || \
+            ! [[ "${{ matrix.installer_version }}" =~ ^([1-9])\.([0-9]+)\.x\-dev$ ]]
+          then
+            # Failed to run composer update and will not attempt requiring different versions of silverstripe/installer
+            cat composer.json
+            cat __update_attempt.txt
+            exit 2
```

Comment:
Why `2`?

Reply:
Same exit code that composer uses when it fails